auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -545,11 +533,10 @@ private IdentitySharedExclusiveLock getRealmIdentityLockForName(final String nam
             this.lock = lock;
             this.hashCharset = hashCharset;
             this.hashEncoding = hashEncoding;
-            this.providers = providers;
+            this.providers = providers != null ? providers : INSTALLED_PROVIDERS;
```

Comment:
The `Identity` is instantiated using parameters from the `FileSystemSecurityRealm` so the existing `providers` value should be used here.

Reply:
Updated

Reply:
It doesn't look like this was upstead. This should be `this.providers = providers`.